File: ChatGPT/c/output/output_09a.c
// ChatGPT-generated solution will go here.
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_SIZE 1000

int main() {
    FILE *f = fopen("input.txt", "r");
    char line[MAX_SIZE];
    fgets(line, sizeof(line), f);
    fclose(f);

    int length = strlen(line);
    int file_id = 0;
    int checksum = 0;

    // Loop through the disk map
    for (int i = 0; i < length; i++) {
        if (line[i] != '.') {
            int file_length = line[i] - '0'; // Get the length of the file
            for (int j = 0; j < file_length; j++) {
                checksum += (i + j) * file_id; // Add to checksum: position * file ID
            }
            file_id++; // Increment file ID
            i += file_length - 1; // Skip over the length of the file
        }
    }

    printf("%d\n", checksum);
    return 0;
}

File: ChatGPT/python/output/output_09a.py
# ChatGPT-generated solution will go here.
def main():
    with open('input.txt') as f:
        line = f.read().strip()

    blocks = []
    file_id = 0
    i = 0
    while i < len(line):
        length = int(line[i])
        is_file = (i % 2 == 0)
        if is_file:
            blocks.extend([str(file_id)] * length)
            file_id += 1
        else:
            blocks.extend(['.'] * length)
        i += 1

    # Compact the files
    i = 0
    while i < len(blocks):
        if blocks[i] == '.':
            j = i + 1
            while j < len(blocks) and blocks[j] == '.':
                j += 1
            if j == len(blocks):
                break
            blocks[i] = blocks[j]
            blocks[j] = '.'
            i = max(i - 1, 0)
        else:
            i += 1

    # Compute checksum
    checksum = 0
    for idx, val in enumerate(blocks):
        if val != '.':
            checksum += idx * int(val)
    print(checksum)

if __name__ == '__main__':
    main()
